lib/types.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+
+export type Range = {
+    start: number,
+    end: number,
+}
+
+
+export type Resolved = {
+    url?: string,
+    filename?: string,
+}
+
+export type Destination = {|
+    start: number,
+    end: number,
+|}
+
+type ExternalModule = {
+    local: string,
+    start: number,
+    end: number,
+    moduleName: string,
+    imported: string,
+}
+
+type SuggestionFromImport = {|
+    type: 'from-import',
+    moduleName: string,
+    imported: string,
+    bindingStart: number,
+    bindingEnd: number,
+|}
+
+type BindingSuggestion = {|
+    type: 'binding',
+    start: number,
+    end: number,
+    moduleName: typeof undefined,
+|}
+
+type PathSuggestion = {|
+    type: 'path',
+    imported: string,
+    moduleName: string,
+    range: Range,
+|}
+
+export type Suggestion =
+    | SuggestionFromImport
+    | BindingSuggestion
+    | PathSuggestion
+
+type Path = {
+    imported: string,
+    moduleName: string,
+    range: Range,
+}
+
+type Binding = {|
+    identifier: Range,
+|}
+type BabelScope = {|
+    block: Range,
+    hasBinding: (string) => boolean,
+    getBinding: (string) => Binding
+|}
+
+type ParseError = {
+    parseError: Error
+}
+export type Info = ParseError | {
+    parseError?: typeof undefined,
```

Comment:
Without this some of [the tests](https://github.com/AsaAyers/js-hyperclick/blob/ecafbbcebbdefd97755aa13c14a3e371fcd03bba/spec/parser-spec.js#L12) complain. I'm not sure if this is really a good solution, but it allows me to continue using `expect()`.

This satisfies Flow, but the `expect()` is useless.
```js
    it('parse-error.js reports a parse error', () => {
        const { code } = extractAnnotations('parse-error.js')
        const info = parseCode(code)

        if (typeof info.parseError == 'undefined') {
            throw new Error(`Expected a parse error`)
        }

        expect(info.parseError).not.toBeUndefined()
    })
```